libcxx/include/__filesystem/directory_iterator.h

```diff
@@ -74,7 +74,8 @@ class directory_iterator {
 
   _LIBCPP_HIDE_FROM_ABI const directory_entry& operator*() const {
     // Note: this check duplicates a check in `__dereference()`.
-    _LIBCPP_ASSERT_NON_NULL(__imp_, "The end iterator cannot be dereferenced");
+    _LIBCPP_REDUNDANT_ASSERTION( //
```

Comment:
The artificial comment is to force `clang-format` not to merge this statement into a single line. I find that having both macros in the same line makes it oversaturated, so to say.